src/command_modules/azure-cli-vm/azure/cli/command_modules/vm/_params.py

```diff
@@ -53,17 +54,31 @@ def get_vm_size_completion_list(prefix, action, parsed_args, **kwargs):  # pylin
-register_cli_argument('vm disk', 'disk_name', options_list=('--name', '-n'), help='The data disk name. If missing, will retrieve from vhd uri')
-register_cli_argument('vm disk', 'disk_size', help='Size of disk (GiB)', default=1023, type=int)
-register_cli_argument('vm disk', 'lun', type=int, help='0-based logical unit number (LUN). Max value depends on the Virutal Machine size.')
-register_cli_argument('vm disk', 'vhd', type=VirtualHardDisk, help='virtual hard disk\'s uri. For example:https://mystorage.blob.core.windows.net/vhds/d1.vhd')
-register_cli_argument('vm disk', 'caching', help='Host caching policy', default=CachingTypes.none.value, **enum_choice_list(CachingTypes))
-
-for item in ['attach-existing', 'attach-new', 'detach']:
-    register_cli_argument('vm disk {}'.format(item), 'vm_name', arg_type=existing_vm_name, options_list=('--vm-name',), id_part=None)
-
-register_cli_argument('vm availability-set', 'availability_set_name', name_arg_type, completer=get_resource_name_completion_list('Microsoft.Compute/availabilitySets'), help='Name of the availability set')
+register_cli_argument('vm unmanaged-disk', 'vm_name', arg_type=existing_vm_name)
+register_cli_argument('vm unmanaged-disk attach', 'disk_name', options_list=('--name', '-n'), help='The data disk name(optional when create a new disk)')
+register_cli_argument('vm unmanaged-disk detach', 'disk_name', options_list=('--name', '-n'), help='The data disk name.')
+register_cli_argument('vm unmanaged-disk', 'disk_size', help='Size of disk (GiB)', default=1023, type=int)
+register_cli_argument('vm unmanaged-disk', 'new', action="store_true", help='create a new disk')
+register_cli_argument('vm unmanaged-disk', 'lun', type=int, help='0-based logical unit number (LUN). Max value depends on the Virutal Machine size.')
+register_cli_argument('vm unmanaged-disk', 'vhd_uri', help="virtual hard disk's uri. For example:https://mystorage.blob.core.windows.net/vhds/d1.vhd")
+register_cli_argument('vm unmanaged-disk', 'caching', help='Host caching policy', default=CachingTypes.none.value, **enum_choice_list(CachingTypes))
+
+for item in ['attach', 'detach']:
+    register_cli_argument('vm unmanaged-disk {}'.format(item), 'vm_name', arg_type=existing_vm_name, options_list=('--vm-name',), id_part=None)
+
+register_cli_argument('vm disk', 'vm_name', options_list=('--vm-name',), id_part=None,
+                      completer=get_resource_name_completion_list('Microsoft.Compute/virtualMachines'))
+register_cli_argument('vm disk', 'disk', validator=validate_vm_disk, help='disk name or id',
+                      completer=get_resource_name_completion_list('Microsoft.Compute/disks'))
+register_cli_argument('vm disk', 'new', action="store_true", help='create a new disk')
+register_cli_argument('vm disk', 'storage_account_type', arg_type=disk_underlined_storage_account_type)
+register_cli_argument('vm disk', 'size_gb', options_list=('--size-gb', '-z'), help='size in GB.')
```

Comment:
let me confirm with CRP folks, this is the term used in swagger, so I prefer to keep.